File: class24/python/AP2411-CalculaMediaPonderada.py
# CalculaMediaPonderada — Aula 24
# Função que calcula média ponderada

def media_ponderada(notas, pesos):
    soma_notas = sum(n * p for n, p in zip(notas, pesos))
    soma_pesos = sum(pesos)
    return soma_notas / soma_pesos if soma_pesos > 0 else 0

notas = [8.5, 7.0, 9.0]
pesos = [2, 3, 5]
media = media_ponderada(notas, pesos)
print(f"Notas: {notas}")
print(f"Pesos: {pesos}")
print(f"Média Ponderada: {media:.2f}")

File: class24/java/AP2411-CalculaMediaPonderada.java
public class CalculaMediaPonderada {
    static double mediaPonderada(double[] notas, int[] pesos) {
        double somaN = 0; int somaP = 0;
        for (int i = 0; i < notas.length; i++) {
            somaN += notas[i] * pesos[i];
            somaP += pesos[i];
        }
        return somaP > 0 ? somaN/somaP : 0;
    }
    public static void main(String[] args) {
        double[] notas = {8.5, 7.0, 9.0};
        int[] pesos = {2, 3, 5};
        System.out.print("Notas: ");
        for (double n : notas) System.out.print(n + " ");
        System.out.print("\nPesos: ");
        for (int p : pesos) System.out.print(p + " ");
        System.out.printf("\nMédia Ponderada: %.2f%n", mediaPonderada(notas, pesos));
    }
}